src/transaction.js

```diff
@@ -61,6 +62,18 @@ export class Transaction {
     }
 
     /**
+     * Add `hashX` signer preimage as signature.
+     * @param {Buffer|String} preimage Preimage of hash used as signer
+     * @returns {void}
+     */
+    signHashX(preimage) {
```

Comment:
I think that if stellar-core processes the transaction with a hash that matches account `hashTx` signer it automatically increases weight of the transaction so there is no need to add another signature to transaction.

But I'm not 100% sure. @jedmccaleb can you confirm?